### DISCUSSION TITLE
Folder overview links didn't appear in the graph view / linked mentions

### DISCUSSION THREAD

**Discussion by Terro216** [LostPaul/obsidian-folder-overview#12] [General]
So if i create folder overview
<details>
<summary>folder</summary>
<img width="653" alt="1" src="https://github.com/LostPaul/obsidian-folder-overview/assets/19672917/e190be9c-39ee-4fbf-bd62-e7dd3791a691">
</details>
It doesn't create real links and shows up in the graph view (and in linked mentions in notes)

<details>
<summary>graph view</summary>
<img width="363" alt="2" src="https://github.com/LostPaul/obsidian-folder-overview/assets/19672917/4fe22285-bd29-4de5-b209-9b5cb342a46f">
</details>

Is it a bug, and can it be fixed (maybe by adding an additional setting for creating links)? Or is it supposed to work this way?

**Comment by gael-vanderlee** [comment]
I'm encountering the same issue.

**Comment by vyragosa** [comment]
Me too

**Comment by mayurankv** [comment]
This is an issue with obsidian not allowing plugins to create links that are properly registered with Obsidian's caches right now. Without a fair amount of work to check what links exist and when they are changed etc. this isn't possible currently. Disclaimer, haven't worked with this plugin but have this issue on one of my plugins.

**Comment by gael-vanderlee** [comment]
@mayurankv Not sure how they do it, but plugins such as [ZotterlKeeper](https://github.com/akosbalasko/zoottelkeeper-obsidian-plugin) have managed to create links that show in the graph view.

**Comment by BramHamaekers** [comment]
@gael-vanderlee Have you discovered how they manage it? I've tried looking at it so I could make a PR for this feature but haven't managed to figure it out myself yet.

**Comment by LostPaul** [comment]
@BramHamaekers I don't specifically know about the mentioned plugin but it probably edits the file itself and adds the links but mine just creates html elements that Obsidian doesn't register

**Comment by dnlbauer** [comment]
[Waypoint](https://github.com/IdreesInc/Waypoint) does it by actually editing the note as well and inserting links as markdown text.

@LostPaul would you be open for PRs implementing similar behaviour in folder notes?
The trickiest part is probably to figure out how to still support the settings folder notes can have inside the link section while upting the notes..

**Reply by LostPaul** [reply]
> [Waypoint](https://github.com/IdreesInc/Waypoint) does it by actually editing the note as well and inserting links as markdown text.
> 
> @LostPaul would you be open for PRs implementing similar behaviour in folder notes? The trickiest part is probably to figure out how to still support the settings folder notes can have inside the link section while upting the notes..


My idea is that you just add the links under the codeblock, update the list when the list of shown files/folders changes in the codeblock and also hide it with css so that you only see the folder overview but also have real links. But currently I'm doing other things first to make the next public release possible. So if you want can create an PR here that implements that behaviour. @dnlbauer

**Comment by LostPaul** [comment]
Has been added to the latest update via the option "Use actual links".
